File: luasrc/model/cbi/ipinfo.lua
local a, t, e
a = Map("ipinfo", translate("IP Information Service"))
a.description = translate(
    "Shows public ip information in Overview LuCI with IP Geolocation API") ..
                    [[<br/><br/><a href="https://github.com/animegasan" target="_blank">Powered by animegasan</a>]]

t = a:section(NamedSection, "config", "ipinfo")
t.anonymous = true
t.addremove = false
--- provider information
t:tab("isp", translate("Provider Information"))
---- ip
e = t:taboption("isp", Flag, "query", translate("Public IP"))
e.default = 0
e.rmempty = false
---- isp
e = t:taboption("isp", Flag, "isp", translate("Provider"))
e.default = 0
e.rmempty = false
---- asname
e = t:taboption("isp", Flag, "asname", translate("AS Name"))
e.default = 0
e.rmempty = false
---- as
e = t:taboption("isp", Flag, "as", translate("AS Number"))
e.default = 0
e.rmempty = false
--- location information
t:tab("loc", translate("Location Information"))
---- city
e = t:taboption("loc", Flag, "city", translate("City"))
e.default = 0
e.rmempty = false
---- regionName
e = t:taboption("loc", Flag, "regionName", translate("Region"))
e.default = 0
e.rmempty = false
---- country
e = t:taboption("loc", Flag, "country", translate("Country"))
e.default = 0
e.rmempty = false
---- continent
e = t:taboption("loc", Flag, "continent", translate("Continent"))
e.default = 0
e.rmempty = false
--- coordinate information
t:tab("co", translate("Coordinate Information"))
---- lat
e = t:taboption("co", Flag, "lat", translate("Latitude"))
e.default = 0
e.rmempty = false
---- lon
e = t:taboption("co", Flag, "lon", translate("Longitude"))
e.default = 0
e.rmempty = false
---- offset
e = t:taboption("co", Flag, "offset", translate("Offset"))
e.default = 0
e.rmempty = false
return a

File: htdocs/luci-static/resources/view/ipinfo.js
/* This is free software, licensed under the Apache License, Version 2.0
 *
 * Copyright (C) 2024 Hilman Maulana <hilman0.0maulana@gmail.com>
 */

'use strict';
'require view';
'require form';

return view.extend({
	render: function () {
		var m, s, o;
		m = new form.Map('ipinfo', _('IP Information'), _('Shows public ip information in Overview LuCI with <a href="https://ip.guide/" target="_blank">ip.guide</a>.'));
		s = m.section(form.TypedSection, 'ipinfo');
		s.anonymous = true;

		o = s.tab('isp', _('Provider Information'));
		o = s.taboption('isp', form.Flag, 'ip', _('Public IP'));
		o.default = '1';
		o.rmempty = false;
		o = s.taboption('isp', form.Flag, 'name', _('Provider'));
		o.default = '1';
		o.rmempty = false;
		o = s.taboption('isp', form.Flag, 'organization', _('Organization'));
		o.default = '1';
		o.rmempty = false;
		o = s.taboption('isp', form.Flag, 'asn', _('AS Number'));
		o.default = '1';
		o.rmempty = false;
		o = s.tab('loc', _('Location Information'));
		o = s.taboption('loc', form.Flag, 'city', _('City'));
		o.default = '1';
		o.rmempty = false;
		o = s.taboption('loc', form.Flag, 'country', _('Country'));
		o.default = '1';
		o.rmempty = false;
		o = s.taboption('loc', form.Flag, 'timezone', _('Timezone'));
		o.default = '1';
		o.rmempty = false;
		o = s.tab('co', _('Coordinate Information'));
		o = s.taboption('co', form.Flag, 'latitude', _('Latitude'));
		o.default = '1';
		o.rmempty = false;
		o = s.taboption('co', form.Flag, 'longitude', _('Longitude'));
		o.default = '1';
		o.rmempty = false;
		return m.render();
	},
});
